devito/profiler.py

```diff
@@ -81,18 +86,23 @@ def add_profiling(self, code, name, byte_size=4, omp_flag=None):
                           Defaults to 4.
         :param omp_flag: OpenMP flag to add before profiling operations,
                          if needed.
+        :param to_ignore: List of strings containing the labels of
+                          symbols used as loop variables
```

Comment:
I don't think a global registry approach would work, since it is quite conceivable to have multiple `Operators` in an application, each with a distinct set of `Iterations` and it's own profiler.